src/main/java/io/fabric8/elasticsearch/plugin/acl/BaseRolesMappingSyncStrategy.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.elasticsearch.plugin.acl;
+
+import static io.fabric8.elasticsearch.plugin.acl.SearchGuardRoles.ROLE_PREFIX;
+
+import java.util.Iterator;
+
+import io.fabric8.elasticsearch.plugin.acl.SearchGuardRolesMapping.RolesMapping;
+
+public abstract class BaseRolesMappingSyncStrategy implements RolesMappingSyncStrategy {
+
+    protected final SearchGuardRolesMapping mappings;
+    
+    protected BaseRolesMappingSyncStrategy(final SearchGuardRolesMapping mappings) {
+        this.mappings = mappings;
+    }
+
+
+    // Remove roles that start with "gen_"
+    private void removeSyncAcls() {
+        for (Iterator<RolesMapping> i = mappings.iterator(); i.hasNext();) {
+            RolesMapping mapping = i.next();
+            if (mapping.getName() != null && mapping.getName().startsWith(ROLE_PREFIX)) {
+                mappings.removeRolesMapping(mapping);
```

Comment:
If you're telling me it's safe then ok.  Usually you'd need to use `.remove()` because [the behavior of an iterator is unspecified if the underlying collection is modified while the iteration is in progress in any way other than by calling this method](https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html).  I don't have the code locally to dig in to so take that with a grain of salt